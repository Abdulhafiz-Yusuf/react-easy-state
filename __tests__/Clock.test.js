import React from 'react'
import { render, cleanup } from 'react-testing-library'
import sinon from 'sinon'
import App from '../examples/clock/src/App'

describe('Clock App', () => {
  const clock = sinon.useFakeTimers()
  const { container } = render(<App />)
  const clearIntervalSpy = sinon.spy(global, 'clearInterval')

  afterAll(() => {
    clock.restore()
    clearIntervalSpy.restore()
  })

  test('should update to display the current time every second', async () => {
    expect(container).toHaveTextContent('12:00:00 AM')

    clock.tick(2000)
    expect(container).toHaveTextContent('12:00:02 AM')

    clock.tick(8500)
    expect(container).toHaveTextContent('12:00:10 AM')
  })

  test('should clean up the interval timer when the component is unmounted', () => {
    cleanup()
    expect(clearIntervalSpy.callCount).toBe(1)
  })
})
