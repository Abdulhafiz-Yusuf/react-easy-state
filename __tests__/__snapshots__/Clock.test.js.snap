// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`01. Initial state 1`] = `
<EasyStateWrapper>
  <div>
    01:00:00 AM
  </div>
</EasyStateWrapper>
------------HTML PREVIEW---------------
<EasyStateWrapper>
  <div>
    01:00:00 AM
  </div>
</EasyStateWrapper>
`;

exports[`02. After 2 seconds 1`] = `
<EasyStateWrapper>
  <div>
    01:00:02 AM
  </div>
</EasyStateWrapper>
------------HTML PREVIEW---------------
<EasyStateWrapper>
  <div>
    01:00:02 AM
  </div>
</EasyStateWrapper>
`;

exports[`03. After 10.5 seconds 1`] = `
<EasyStateWrapper>
  <div>
    01:00:10 AM
  </div>
</EasyStateWrapper>
------------HTML PREVIEW---------------
<EasyStateWrapper>
  <div>
    01:00:10 AM
  </div>
</EasyStateWrapper>
`;
